api/api/controllers/agent_controller.py

```diff
@@ -84,71 +140,40 @@ async def get_agents(pretty: bool = False, wait_for_complete: bool = False, agen
     sort : str
         Sort the collection by a field or fields (separated by comma). Use +/- at the beginning to list in
         ascending or descending order.
-    search : str
-        Look for elements with the specified string.
-    status : str
-        Filter by agent status. Use commas to enter multiple statuses.
-    q : str
-        Query to filter results by. For example "q&#x3D;&amp;quot;status&#x3D;active&amp;quot;".
-    older_than : str
-        Filter out disconnected agents for longer than specified. Time in seconds, ‘[n_days]d’,
-        ‘[n_hours]h’, ‘[n_minutes]m’ or ‘[n_seconds]s’. For never_connected agents, use the register date.
-    manager : str
-        Filter by manager hostname to which agents are connected.
-    version : str
-        Filter by agents version.
-    group : str
-        Filter by agent group.
-    node_name : str
-        Filter by node name.
-    name : str
-        Filter by agent name.
-    ip : str
-        Filter by agent IP.
-    group_config_status : str
-        Filter by agent groups configuration sync status.
-    distinct : bool
-        Look for distinct values.
 
     Returns
     -------
     ConnexionResponse
         Response with all selected agents' information.
     """
-    f_kwargs = {'agent_list': agents_list,
-                'offset': offset,
-                'limit': limit,
-                'sort': parse_api_param(sort, 'sort'),
-                'search': parse_api_param(search, 'search'),
-                'select': select,
-                'filters': {
-                    'status': status,
-                    'older_than': older_than,
-                    'manager': manager,
-                    'version': version,
-                    'group': group,
-                    'node_name': node_name,
-                    'name': name,
-                    'ip': ip,
-                    'registerIP': request.query_params.get('registerIP', None),
-                    'group_config_status': group_config_status
-                },
-                'q': q,
-                'distinct': distinct
-                }
-    # Add nested fields to kwargs filters
-    nested = ['os.version', 'os.name', 'os.platform']
-    for field in nested:
-        f_kwargs['filters'][field] = request.query_params.get(field, None)
+    if older_than is not None and older_than.isnumeric():
+        older_than = f'{older_than}s'
+
+    f_kwargs = {
+        'agents_id': agents_id if agents_id is not None else [],
```

Comment:
To be consistent with the rest of the code we could use `agent_list`, otherwise I would prefer `agent_ids`.

```suggestion
        'agent_list': agent_list if agent_list is not None else [],
```

Reply:
I decided to go with `agents_list` to be consistent with the query parameter that receives the three endpoints.